### ISSUE TITLE
res is undefined when http call fails at parsePackage - FIX

### ISSUE THREAD

**Issue by nombrekeff** [jesseditson/npm-package#1]
Hi, 
I've come across a little bug when there is no internet conection, caused by checking for `res.status` before checking the **error** at:
https://github.com/jesseditson/npm-package/blob/3329109fff2902de062ff7719d4bb029a6bb64ab/lib/parse-package.js#L5

#### Bug description:
So I didn't have internet access when the bug hapened, so I expected to throw some error.
The proplem is that it throws: 
```
C:\Users\keff\Desktop\cosas\Dev\LibsJS\npm-info\node_modules\npm-package-info\lib\parse-package.js:5
    switch (res.status) {
                ^

TypeError: Cannot read property 'status' of undefined
    at C:\Users\keff\Desktop\cosas\Dev\LibsJS\npm-info\node_modules\npm-package-info\lib\parse-package.js:5:17
```

I checked the code and the problem is as explained above, that you are checking for the `status` before checking for errors, so if an error ocurres `res ` is undefined so it throws the error shown above.

I would sugest doing the following
```js
function parsePackage(callback) {
  return function(err, res) {
    if (err) return callback(err) // Move error checking to first line
    
   // And after error checking you are good to check the response
    switch (res.status) {
      case 404:
        return callback(new Error('Unable to find package.'))
    }

    var pkg = res.body
    try {
      pkg = JSON.parse(res.text)
    } catch(e) {
      // falling back to res.body
    }
    callback(null, pkg)
  }
}
```

Move this line to above the switch:
https://github.com/jesseditson/npm-package/blob/3329109fff2902de062ff7719d4bb029a6bb64ab/lib/parse-package.js#L9

**Comment by jesseditson** [comment]
Thanks for the report! I’ll update when I get a chance, or if you feel like it i’ll happily accept a PR and publish that change!

**Comment by nombrekeff** [comment]
Ohh nice! Wasn't expecting an answer so rapidly!
I was thinking on doing a PR but I prefered to tell you about it first!

I will fix it tonight in a moment!

PD:
Would you be interested in me collaborating in this project? Im looking for projects to collaborate and can't find many! And this would be a cool one! It has some potential! If we do it right ;)...

**Comment by jesseditson** [comment]
Resolved in #2, published as npm-package-info@0.0.5
